python-squid/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -869,31 +869,40 @@ public void classdef_statement() {
   public void for_statement() {
     setRootRule(PythonGrammar.FOR_STMT);
     AstNode astNode = p.parse("for foo in bar: pass");
-    ForStatement pyForStatementTree = treeMaker.forStatement(astNode);
-    assertThat(pyForStatementTree.expressions()).hasSize(1);
-    assertThat(pyForStatementTree.testExpressions()).hasSize(1);
-    assertThat(pyForStatementTree.body().statements()).hasSize(1);
-    assertThat(pyForStatementTree.body().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
-    assertThat(pyForStatementTree.elseBody()).isNull();
-    assertThat(pyForStatementTree.isAsync()).isFalse();
-    assertThat(pyForStatementTree.asyncKeyword()).isNull();
-    assertThat(pyForStatementTree.children()).hasSize(7);
+    ForStatement forStatement = treeMaker.forStatement(astNode);
+    assertThat(forStatement.expressions()).hasSize(1);
+    assertThat(forStatement.testExpressions()).hasSize(1);
+    assertThat(forStatement.body().statements()).hasSize(1);
+    assertThat(forStatement.body().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
+    assertThat(forStatement.elseBody()).isNull();
+    assertThat(forStatement.isAsync()).isFalse();
+    assertThat(forStatement.asyncKeyword()).isNull();
+    assertThat(forStatement.children()).hasSize(7);
+
+    astNode = p.parse("for foo in bar:\n  pass\n");
+    forStatement = treeMaker.forStatement(astNode);
+    assertThat(forStatement.expressions()).hasSize(1);
+    assertThat(forStatement.testExpressions()).hasSize(1);
+    assertThat(forStatement.body().statements()).hasSize(1);
+    assertThat(forStatement.body().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
+    assertThat(forStatement.elseBody()).isNull();
+    assertThat(forStatement.children()).hasSize(10);
 
     astNode = p.parse("for foo in bar:\n  pass\nelse:\n  pass");
-    pyForStatementTree = treeMaker.forStatement(astNode);
-    assertThat(pyForStatementTree.expressions()).hasSize(1);
-    assertThat(pyForStatementTree.testExpressions()).hasSize(1);
-    assertThat(pyForStatementTree.body().statements()).hasSize(1);
-    assertThat(pyForStatementTree.body().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
-    assertThat(pyForStatementTree.elseBody().statements()).hasSize(1);
-    assertThat(pyForStatementTree.elseBody().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
-    assertThat(pyForStatementTree.children()).hasSize(15);
+    forStatement = treeMaker.forStatement(astNode);
+    assertThat(forStatement.expressions()).hasSize(1);
+    assertThat(forStatement.testExpressions()).hasSize(1);
+    assertThat(forStatement.body().statements()).hasSize(1);
+    assertThat(forStatement.body().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
+    assertThat(forStatement.elseBody().statements()).hasSize(1);
+    assertThat(forStatement.elseBody().statements().get(0).is(Tree.Kind.PASS_STMT)).isTrue();
+    assertThat(forStatement.children()).hasSize(15);
```

Comment:
I think it would be nice to add assertions for `newline`, `indent` and `dedent` tokens to make it more explicit